framework/src/controller/controller.js

```diff
@@ -203,23 +213,42 @@ class Controller {
 			`Loading module with alias: ${moduleAlias}(${name}:${version})`
 		);
 
-		const channel = new ChildProcessChannel(
-			moduleAlias,
-			module.events,
-			module.actions
+		const modulePath = path.resolve(
+			__dirname,
+			'../modules',
+			alias.replace(/([A-Z])/g, $1 => `_${$1.toLowerCase()}`)
 		);
 
-		await channel.registerToBus(this.socketsPath);
+		const program = path.resolve(__dirname, 'child_process_loader.js');
 
-		channel.publish(`${moduleAlias}:registeredToBus`);
-		channel.publish(`${moduleAlias}:loading:started`);
-		await module.load(channel);
-		channel.publish(`${moduleAlias}:loading:finished`);
+		const parameters = [
+			modulePath,
+			JSON.stringify({ config: this.config, moduleOptions: options }),
+		];
 
-		this.modules[moduleAlias] = module;
-		this.logger.info(
-			`Module ready with alias: ${moduleAlias}(${name}:${version})`
-		);
+		const child = child_process.fork(program, parameters);
+
+		child.on('exit', (code, signal) => {
+			this.logger.error(
+				`Module ${moduleAlias}(${name}:${version}) exited with code: ${code} and signal: ${signal}`
+			);
+			// TODO: Reload child instead of exiting the parent process
+			process.exit(1);
```

Comment:
Is this going to be fixed in another PR?

Reply:
https://github.com/LiskHQ/lisk/issues/3142